fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/UnbakedModelDeserializer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.model.loading.v1;
+
+import java.io.Reader;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.client.render.model.UnbakedModel;
+import net.minecraft.client.render.model.json.JsonUnbakedModel;
+import net.minecraft.client.render.model.json.ModelElement;
+import net.minecraft.client.render.model.json.ModelElementFace;
+import net.minecraft.client.render.model.json.ModelElementTexture;
+import net.minecraft.client.render.model.json.ModelTransformation;
+import net.minecraft.client.render.model.json.Transformation;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.client.model.loading.UnbakedModelDeserializerRegistry;
+
+/**
+ * Allows creating custom unbaked models by overriding the parsing of JSON model files.
+ *
+ * <p>The format for custom unbaked models is as follows:
+ * <pre>{@code
+ * {
+ *     "fabric:type": "<identifier of the deserializer>",
+ *     // extra model data, dependent on the deserializer
+ * }
+ * }</pre>
+ *
+ * <p>Alternatively, {@code "fabric:type"} may be an object with the required string field {@code "id"}, specifying the
+ * identifier of the deserializer, and the optional boolean field {@code "optional"} with default {@code false},
+ * specifying whether the model should fail loading ({@code false}) or continue loading as a vanilla model
+ * ({@code true}) when the specified deserializer has not been registered.
+ *
+ * <p>All instances must be registered using {@link #register} for deserialization to work.
+ */
+public interface UnbakedModelDeserializer {
+	/**
+	 * Registers a custom model deserializer.
+	 *
+	 * @throws IllegalArgumentException if the deserializer is already registered
+	 */
+	static void register(Identifier id, UnbakedModelDeserializer deserializer) {
+		UnbakedModelDeserializerRegistry.register(id, deserializer);
+	}
+
+	/**
+	 * @return the custom model deserializer registered with the given identifier, or {@code null} if there is no such
+	 * deserializer
```

Comment:
```suggestion
	 * {@return the custom model deserializer registered with the given identifier, or {@code null} if there is no such
	 * deserializer}
```

Reply:
What is the purpose of this change?

Reply:
It renders in a nicer way. From the [spec](https://docs.oracle.com/en/java/javase/23/docs/specs/javadoc/doc-comment-spec.html#return):
> As an inline tag, provides content for the first sentence of a method's main description, and a "Returns" section, as if "`@return description`" were also present. In the default English locale, the first sentence is "Returns *description*.".

I don't think there's any description for the method in the method list if you use the block tag form (`@return` without braces).

Reply:
Only `@return` with no text before means that there won't be any text in the summary of the method. I personally prefer not using any tag in that case.

Reply:
The whole point of the inline `{@return}` is that it also adds a short method summary ("Returns ....") as well as the return value section. IMO only `@return` < no tag ≤ only `{@return}`, but this is very bikesheddy and anything works in practice.